pkg/kudoctl/cmd/init/prereqs.go

```diff
@@ -223,3 +236,69 @@ func namespace(namespace string) *v1.Namespace {
 	}
 	return ns
 }
+
+// Validate whether the serviceAccount exists
+func validateServiceAccountExists(client corev1.ServiceAccountsGetter, opts Options) error {
+
+	serviceAccountExists := false
+
+	saList, err := client.ServiceAccounts(opts.Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, sa := range saList.Items {
+		if sa.Name == opts.ServiceAccount {
+			serviceAccountExists = true
+			break
+		}
+	}
+
+	if !serviceAccountExists {
+		return fmt.Errorf("Service Account %s does not exists - KUDO expects the serviceAccount to be present", opts.ServiceAccount)
+	}
+	return nil
+}
+
+// Validate whether the serviceAccount has cluster-admin role
+func validateClusterAdminRoleForSA(client kubernetes.Interface, opts Options) error {
+
+	saCheck := false
+	nsCheck := false
+	crCheck := false
+
+	// Check whether the serviceAccount has clusterrolebinding cluster-admin
+	crbs, err := client.RbacV1().ClusterRoleBindings().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, crb := range crbs.Items {
+		// Check whether the serviceAccount is part of the cluster-admin clusterrolebindings Subject's
+		for _, subject := range crb.Subjects {
+			if subject.Name == opts.ServiceAccount {
+				saCheck = true
+				if subject.Namespace == opts.Namespace {
+					nsCheck = true
+					if crb.RoleRef.Name == "cluster-admin" {
+						crCheck = true
+						break
+					}
+				}
+			}
+		}
+	}
```

Comment:
i had to think a lot about this block. while i've convinced myself we need the nested ifs for now to not throw confusing errors (preferring user experience > readable code > optimization, where I have a different answer for each level), I still have to think about this one for a possible later refactoring.

Reply:
While testing I noticed K8S systems accepts all combinations and does not throw any error at all, so, I thought it is much more user friendly to be as explicit with the error as possible.